File: Data Structures & Algorithms/count-number-of-islands/submission-0.java
class Solution {
    public int numIslands(char[][] grid) {
        int num = 0;

        for(int i = 0; i < grid.length; i++){
            for(int j = 0; j < grid[0].length; j++){
                if(grid[i][j] == '1'){
                    num++;
                    dfs(grid, i, j);
                }
            }
        }

        return num;
    }

    public void dfs(char[][] grid, int row, int col){
        int[] location = {-1, 0, 1, 0, -1};

        grid[row][col] = '0';

        for(int i = 0; i < location.length - 1; i++){
            int currRow = row + location[i];
            int currCol = col + location[i+1];

            if(currRow >= 0 && currRow < grid.length && currCol >=0 && currCol < grid[0].length && grid[currRow][currCol] == '1'){
                dfs(grid, currRow, currCol);
            }
        }
    }
}

File: Data Structures & Algorithms/islands-and-treasure/submission-0.py
class Solution:
    def islandsAndTreasure(self, grid: List[List[int]]) -> None:

        def dfs(grid : List[List[int]], i : int, j : int, dist : int) -> None:
            dist += 1
            locations = [(-1, 0), (0, 1), (1, 0), (0, -1)]

            for x,y in locations:
                curr_row = i + x
                curr_col = j + y

                if 0 <= curr_row < len(grid) and 0 <= curr_col < len(grid[0]) and grid[curr_row][curr_col] > 0:
                    if grid[curr_row][curr_col] > dist:
                        grid[curr_row][curr_col] = dist
                        dfs(grid, curr_row, curr_col, dist)

        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == 0:
                    dfs(grid, i, j, 0)
        